src/app/charts/chart.component.ts

```diff
@@ -76,24 +81,46 @@ export class ChartComponent implements OnChanges {
                 private indicatorService: IndicatorService,
                 private dataExportService: DataExportService,
                 private imageExportService: ImageExportService,
-                private authService: AuthService) {}
+                private authService: AuthService,
+                private changeDetector: ChangeDetectorRef) {}
+
+    ngAfterViewInit() {
+        // Manually trigger change detection in parent to avoid
+        // ExpressionChangedAfterItHasBeenCheckedError when extra params form
+        // calls to set initial values in ngAfterViewInit.
+        this.changeDetector.detectChanges();
+    }
 
-    ngOnChanges() {
+    ngOnChanges($event) {
         if (!this.scenario || !this.city || !this.models) { return; }
+        this.updateChart($event);
+    }
+
+    updateChart(extraParams: any) {
         this.chartData = [];
         this.rawChartData = [];
+
+        let params = {
+            climateModels: this.models,
+            unit: this.unit || this.chart.indicator.default_units,
+            // TODO: #212
+            // As a temporary solution, the time agg defaults to the 1st valid option.
+            // Really, this should a user selectable option
+            time_aggregation: this.chart.indicator.valid_aggregations[0]
+        }
+
+        if (isThresholdIndicator(this.chart.indicator.name)) {
+            params = _.extend(params, extraParams);
+            this.isThresholdIndicator = true;
```

Comment:
Hm actually, this only resolves switching between threshold and non-threshold indicators. I still get unwanted persistence this way when switching between threshold indicators..